src/coreclr/tools/Common/TypeSystem/Common/MetadataFieldLayoutAlgorithm.cs

```diff
@@ -993,6 +993,11 @@ private static SizeAndAlignment ComputeInstanceSize(MetadataType type, LayoutInt
                 }
             }
 
+            if (type.ContainsGCPointers)
```

Comment:
For reference, here is the matching logic in the runtime type loader:

https://github.com/dotnet/runtime/blob/8c68710084b861c44b6868c0bd972ce24aed55dd/src/coreclr/vm/methodtablebuilder.cpp#L8868-L8872